packages/agent-utils/src/llm/ChatLogs.ts

```diff
@@ -0,0 +1,76 @@
+import { ChatCompletionRequestMessage as ChatMessage } from "openai";
+
+export { ChatMessage };
+
+export type ChatLogType =
+  | "persistent"
+  | "temporary";
+
+export interface ChatLog {
+  tokens: number;
+  msgs: ChatMessage[];
+}
+
+export class ChatLogs {
+  private _logs: Record<ChatLogType, ChatLog> = {
+    "persistent": {
+      tokens: 0,
+      msgs: [],
+    },
+    "temporary": {
+      tokens: 0,
+      msgs: [],
+    },
+  };
+
+  constructor(logs?: Record<ChatLogType, ChatLog>) {
+    if (logs) {
+      this._logs = logs;
+    }
+  }
+
+  get tokens(): number {
+    return this._logs["persistent"].tokens +
+      this._logs["temporary"].tokens;
+  }
+
+  get messages(): ChatMessage[] {
+    return [
+      ...this._logs["persistent"].msgs,
+      ...this._logs["temporary"].msgs
+    ];
+  }
+
+  static from(chatLog: ChatLog): ChatLogs {
```

Comment:
This could be misleading. The user doesn't realize that ChatLog.from(log) adds the log to persistent messages.
Also if the goal is to create a log from "simple" messages, then having an overload where you don't need to manually specify the token count would be nice

Reply:
Removed this to avoid the ambiguity.